### PR TITLE
[lldb-moduleimport-test] Refactor dump of types from mangled name.

### PR DESCRIPTION
**Pull request by dcci**
In preparation for adding support for dump decl from mangled name.

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci please smoke test and merge

File: tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp [hunk 1]
```diff
@@ -62,6 +62,33 @@ static void printValidationInfo(llvm::StringRef data) {
   }
 }
 
+static void resolveTypeFromMangledNameList(
+    swift::ASTContext &Ctx, llvm::SmallVector<std::string, 8> &MangledNames) {
```
Comment (jrose-apple) [inline tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp]: LLVM style nitpicks: ArrayRef here…

File: tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp [hunk 2]
```diff
@@ -62,6 +62,33 @@ static void printValidationInfo(llvm::StringRef data) {
...
+    } else {
+      ResolvedType->print(llvm::errs());
+      llvm::errs() << "\n";
+    }
+  }
+}
+
+static void
+collectMangledNames(std::string &FilePath,
+                    llvm::SmallVector<std::string, 8> &MangledNames) {
```
Comment (jrose-apple) [inline tools/lldb-moduleimport-test/lldb-moduleimport-test.cpp]: …and SmallVectorImpl here (don't need to hardcode the size). Also, `const` on `FilePath`.

**Comment by dcci** [comment]
Thanks for your review Jordan, I'll include these fixes in my next pull request.

**Pull request merged**